projects/js-packages/publicize-components/src/hooks/use-is-resharing-possible/index.ts

```diff
@@ -0,0 +1,35 @@
+import { useSelect } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
+import { useIsSharingPossible } from '../use-is-sharing-possible';
+import usePublicizeConfig from '../use-publicize-config';
+import useSharePost from '../use-share-post';
+
+/**
+ * Returns whether re-sharing is possible currently.
+ *
+ * False when
+ * - sharing is disabled
+ * - no enabled connections
+ * - post is not published
+ * - is sharing post
+ *
+ * @return {boolean} Whether re-sharing is possible.
+ */
+export function useIsReSharingPossible() {
+	const { isPublicizeEnabled } = usePublicizeConfig();
+	const isSharingPossible = useIsSharingPossible();
+	const { isFetching } = useSharePost();
+
+	const { isPostPublished, isSavingPost } = useSelect( select => {
+		const editorSelector = select( editorStore );
+
+		return {
+			isPostPublished: editorSelector.isCurrentPostPublished(),
+			isSavingPost: editorSelector.isSavingPost(),
+		};
+	}, [] );
+
+	return (
+		isPublicizeEnabled && isSharingPossible && isPostPublished && ! isFetching && ! isSavingPost
+	);
```

Comment:
Not related to this PR but I recently saw this pattern that when you pass the store to `useSelect` instead of the callback, it gives you the selectors which you can call. It makes the code much cleaner, I think. We can use the callback when we need to do some computation inside that callback.

```suggestion
	const { isCurrentPostPublished: isPostPublished, isSavingPost } = useSelect( editorStore, [] );

	return (
		isPublicizeEnabled && isSharingPossible && ! isFetching && isPostPublished() && ! isSavingPost()
	);
```

Reply:
Yeah good point. I usually copy the selectors over from other files that's where this came from, but yeah that should be cleaner, thanks